### PR TITLE
fix(cico): fix back and cancel buttons on Review Screen

### PR DESCRIPTION
**Pull request by jh2oman**
### Description

Talked to Nitya and we opted not to do the popup when the user clicks 'cancel'

https://user-images.githubusercontent.com/8432644/182503257-529e013f-3bfe-43e9-9c1b-9ca0cb873179.mp4

### Tested

manual walkthrough

### How others should test

1] Select celo Dollar
2] Tap on next
3] Enter valid amount and proceed to select payment page
4] Click on bank account from select payment page
5] Now enter bank details and proceed to review page
6] Now from review page click on < arrow
7] It will redirect to Select Payment page

And for cancel button

1] Select celo Dollar
2] Tap on next
3] Enter valid amount and proceed to select payment page
4] Click on bank account from select payment page
5] Now enter bank details and proceed to review page
6] Now from review page click Cancel
7] It will redirect to Add / Withdraw / Spend screen

### Other Changes

Renamed some analytics events to bring them more in line with our fiatconnect standard

### Related issues

- Fixes https://github.com/valora-inc/wallet/issues/2724

### CODE REVIEW

File: src/fiatconnect/ReviewScreen.tsx [hunk 1]
```diff
@@ -343,11 +372,14 @@ FiatConnectReviewScreen.navigationOptions = ({
   ...emptyHeader,
   headerLeft: () => <BackButton />,
   // NOTE: copies for cash in not final
+
   headerTitle:
     route.params.flow === CICOFlow.CashIn
       ? i18n.t(`fiatConnectReviewScreen.cashIn.header`)
       : i18n.t(`fiatConnectReviewScreen.cashOut.header`),
   // TODO(any): when tying this component to the flow, add `onCancel` prop to
   // navigate to correct screen.
```
Comment (satish-ravi) [inline src/fiatconnect/ReviewScreen.tsx:381]: remove?

File: src/fiatconnect/ReviewScreen.tsx [hunk 2]
```diff
@@ -343,11 +372,14 @@ FiatConnectReviewScreen.navigationOptions = ({
...
-  headerRight: () => <CancelButton style={styles.cancelBtn} />,
+  headerRight: () => (
+    <CancelButton onCancel={() => navigate(Screens.FiatExchange)} style={styles.cancelBtn} />
```
Comment (satish-ravi) [inline src/fiatconnect/ReviewScreen.tsx:383]: Do we need analytics for this?

File: src/fiatconnect/ReviewScreen.tsx [hunk 3]
```diff
@@ -41,12 +41,41 @@ export default function FiatConnectReviewScreen({ route, navigation }: Props) {
...
+      goBack()
+      return true
+    }
+    const backHandler = BackHandler.addEventListener('hardwareBackPress', hardwareBackPress)
+    return function cleanup() {
+      backHandler.remove()
+    }
+  }, [])
+
+  const goBack = () => {
```
Comment (satish-ravi) [inline src/fiatconnect/ReviewScreen.tsx:55]: Could this just be part of `onPressBack`? Or do we not want analytics to fire on pressing hardware back?

Comment (jh2oman) [reply src/fiatconnect/ReviewScreen.tsx:55]: We don't fire events for hardware back button anywhere else in the app, so to be consistent I didn't think we should here.

**Review by satish-ravi — APPROVED** [review]

**Review by jophish — APPROVED** [review]

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the above issue on Latest iOS Test Flight Build V:1.38.0 and updates status in test rails

**Observation :**  
- Clicking on back arrow from review page is redirecting to select withdraw payment page and again clicking on it redirects to review page
- If user has turned off internet and clicks on withdraw funds button, then on try again page the blue banner does not go even after clicking on refresh button due to cancel withdraw button is getting overlapped (iOS Specific) and we raised bug for it

**New bug link :** [bug](https://github.com/valora-inc/wallet/issues/2768)

**Test Rails link:**
**iOS 14:** https://valoraapp.testrail.io/index.php?/tests/view/57039

**Device :** iPhone 12(14.7.1)

**CC:** @jh2oman 

Thanks..!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above issue on Android Internal Release Build V:1.38.0 and updated status in test rails


Test Rails link:

Android 11: https://valoraapp.testrail.io/index.php?/tests/view/57053
Device : Google Pixel 2XL (11.0)

CC: @jh2oman
Thanks..!!

**Pull request merged**